src/components/autocomplete/js/autocompleteController.js

```diff
@@ -466,9 +466,9 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         select(ctrl.index);
         break;
       case $mdConstant.KEY_CODE.ESCAPE:
+        event.preventDefault(); // Prevent browser from always clearing input
```

Comment:
I figured out why the tests are passing as is. Typing something would trigger the autocomplete window. Pressing escape closes it and wouldn't clear the text. That's because, with the pop-up open, ‘shouldProcessEscape‘ passes.

I should do a SECOND escape press which would cause ‘shouldProcessEscape‘ to return false (popup is closed) and then the default browser action would kick in.

I'll add the test to this PR, where it should pass and master should fail.